### PR TITLE
Easy image upload refactoring

### PR DESCRIPTION
**Pull request by mlewand**
## What is the purpose of this pull request?

Other, refactoring of other PR.

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](http://docs.ckeditor.com/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](http://docs.ckeditor.com/#!/guide/dev_tests) and
[how to create tests](http://docs.ckeditor.com/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

This PR adds some fixes to #1052, the general idea was to simplify it and extract cloud-service (backend) stuff to a separate plugin.

There were already quite a few changes, many of which has been already fixed in #1062 and #1066.

Summary for this PR:

* [`uploadUrl`](https://github.com/ckeditor/ckeditor-dev/blob/de017b395220ebaba8136ef7650d54dd035b5411/plugins/easyimage/plugin.js#L137) is now optional, as the plugin uses CloudService loader which automatically picks up URL from the config.
* [`loadMethod`](https://github.com/ckeditor/ckeditor-dev/blob/de017b395220ebaba8136ef7650d54dd035b5411/plugins/easyimage/plugin.js?utf8=%E2%9C%93#L139) is now changed to upload. Since you proposed a more modern approach with object URL - I think it's fine, however we need a followup ticket to keep track of data allocated with it. See TP#3131.
* Removed `fileUploadResponse`, `fileUploadRequest` as they're no longer needed (fixed in #1062, #1066).
* [paste listener](https://github.com/ckeditor/ckeditor-dev/blob/4c7e90d472cc1490df1963c6eb158914e8fefbe4/plugins/easyimage/plugin.js#L206) for `uploadeasyimage` widget uses widget definition at a time of paste event to avoid #1068. Though still I'm not happy that it's a copy pasta of a listener from `imageupload` plugin, we need to combine it. See TP#3132.
* Config options `easyimageUploadUrl`, `easyimage_token` have been renamed to `cloudServices_url`, `cloudServices_token` respectively.
* Corrected [`responseData` structure](https://github.com/ckeditor/ckeditor-dev/blob/4c7e90d472cc1490df1963c6eb158914e8fefbe4/tests/plugins/easyimage/uploadwidget.js?utf8=%E2%9C%93#L61-L66) in unit tests.

There's one issue that latest FF is extremely unstable, that's something that we need to investigate in a followup ticket.

Once this is merged, I'll just give a final check to #1052 and that will be it in terms of uploading.

### CODE REVIEW

File: tests/plugins/easyimage/uploadwidget.js [hunk 1]
```diff
@@ -438,6 +430,30 @@
 			} );
 
 			wait();
+		},
+
+		'test cloudservices URL can be customized': function( editor ) {
```
Comment (Comandeer) [inline tests/plugins/easyimage/uploadwidget.js]: Not only URL can be customised, but also loader type and additional request parameters. It'd be nice to have also tests for these.

**Comment by mlewand** [comment]
For reasons unknown to me firefox fires `8` [img load events](https://github.com/ckeditor/ckeditor-dev/blob/77dd5c9609672843acbb07b427a9266e56ec9847/tests/plugins/uploadfile/_helpers/waitForImage.js#L8) in vs `6` in other browsers.

Without getting into details I have changed listener type to `once` instead.

This test suite relies on `img#load` event, with `img[src="data:image/png;base64,..."]` value, so it might hit some quirky edge cases.

Also I have added requested test cases.
